src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Setting/SettingsManagerWrapper.cs

```diff
@@ -1,52 +1,62 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
 using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using NuGet.VisualStudio;
 
 namespace NuGet.PackageManagement.VisualStudio
 {
     internal class SettingsManagerWrapper : ISettingsManager
     {
-        private readonly IVsSettingsManager _settingsManager;
+        private readonly AsyncLazy<IVsSettingsManager> _settingsManager;
 
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
         public SettingsManagerWrapper(IServiceProvider serviceProvider)
         {
-            _settingsManager = (IVsSettingsManager)serviceProvider.GetService(typeof(SVsSettingsManager));
-            Debug.Assert(_settingsManager != null);
+            _settingsManager = new AsyncLazy<IVsSettingsManager>(async () =>
+            {
+                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                return (IVsSettingsManager)serviceProvider.GetService(typeof(SVsSettingsManager));
```

Comment:
>serviceProvider.GetService [](start = 43, length = 26)

Can you use `GetServiceAsync` here instead? the settings manager can be expensive and it would be good to avoid blocking the UI thread while it is constructed. You'll need to either query for IAsyncServiceProvider from your IServiceProvider or just pass IAsyncServiceProvider in to begin with.